Api documentation changes by Praveen

priyanka-TL opened the pull request.

praveenKDass committed:
api documentation changes

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -1,10 +1,12 @@
   description: |-
     - The Survey Project Creation Portal Service is a centralized service that supports designing of resources.
     - The URL to access the Creation Portal API(s) is `{context}/scp/v1` - <b>Note:</b> These resources can be used in other services.
+    - For all CRUD operation using  `SYSTEMADMIN` pass <b>tenant-id</b> and <b>organization-id</b> in header for `ORGADMIN` backend will take it from token
```
@praveenKDass Check the /scp/v1/certificates/list api response in dev and update apidoc accordingly

praveenKDass replied:
Added @priyanka-TL

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -1,10 +1,12 @@
     - The Survey Project Creation Portal Service is a centralized service that supports designing of resources.
     - The URL to access the Creation Portal API(s) is `{context}/scp/v1` - <b>Note:</b> These resources can be used in other services.
+    - For all CRUD operation using  `SYSTEMADMIN` pass <b>tenant-id</b> and <b>organization-id</b> in header for `ORGADMIN` backend will take it from token
+    _One Majpr change in this version how we get certificate `base_template_url` in project Details and program Details, Response structure Attached in project Details API RESPONSE
```
@praveenKDass Please update the /scp/v1/config/list api wrt new response

praveenKDass replied:
Updated @priyanka-TL

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -2786,7 +2788,7 @@ paths:
                       description: The ID of the base certificate template.
                     base_template_url:
-                      type: string
+                      type: object
```
@praveenKDas please check projects/details api response the tenant_code and all missing, please cross check all api

praveenKDass replied:
Addressed @priyanka-TL

priyanka-TL reviewed (commented):
Please recheck all the responses and add missing keys

praveenKDass committed:
api docs updated

praveenKDass committed:
api docs updated with mitra as a task

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -5,6 +5,8 @@ info:
     - The Survey Project Creation Portal Service is a centralized service that supports designing of resources.
     - The URL to access the Creation Portal API(s) is `{context}/scp/v1` - <b>Note:</b> These resources can be used in other services.
+    - For all CRUD operation using  `SYSTEMADMIN` pass <b>tenant-id</b> and <b>organization-id</b> in header for `ORGADMIN` backend will take it from token
+    _One Majpr change in this version how we get certificate `base_template_url` in project Details and program Details, Response structure Attached in project Details API RESPONSE
```
@praveenKDass This line we can skip actually

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -1242,7 +1244,9 @@ paths:
                         type: string
                       organization_code:
-                        type: number
+                        type: string
```
@praveenKDass Check and update entityType/Create api response

priyanka-TL replied:
@praveenKDass Check and update entityType/read api response

priyanka-TL replied:
@praveenKDass Check and update entityType/update api response

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -1242,7 +1244,9 @@ paths:
                       organization_code:
-                        type: number
+                        type: string
+                      tenant_code:
```
@praveenKDass L Update the response of v1/entities/create, v1/entities/update, /v1/entities/read, v1/entities/delete, /v1/entities/list

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -1242,7 +1244,9 @@ paths:
-                        type: number
+                        type: string
+                      tenant_code:
+                        type: string
```
@praveenKDass check all form apis and update the response

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -1385,7 +1389,8 @@ paths:
                         data_type: STRING
                         has_entities: true
-                        organization_code: 1
+                        organization_code: default_code
```
@praveenKDass project creation request, reflection as a task, link is present but button text is not prseent add that

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -1385,7 +1389,8 @@ paths:
                         has_entities: true
-                        organization_code: 1
+                        organization_code: default_code
+                        tenant_code: default
```
@praveenKDass project detail api response with comments  and update the response

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -2232,7 +2237,9 @@ paths:
                         type: string
                       organization_code:
-                        type: number
+                        type: string
```
@praveenKDass check in v1/resource/list api we are passing tenant_code or not

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -2232,7 +2237,9 @@ paths:
                       organization_code:
-                        type: number
+                        type: string
+                      tenant_code:
```
@praveenKDass Check the rollout/details api and add the missing keys, update targeting_criteria, please note the current req  of targeting_criteria we can keep as sunbird request body

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -2232,7 +2237,9 @@ paths:
-                        type: number
+                        type: string
+                      tenant_code:
+                        type: string
```
@praveenKDass check rollout/create api, update api,  request update the targeting_criteria, please note the current req we can keep as sunbird request body

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -2280,7 +2287,7 @@ paths:
                       created_at: '2023-07-18T20:48:10.492Z'
                       deleted_at: null
-                      organization_code: 1
+                      organiation_code: default_code
```
@praveenKDass v1/programs/update, current req we can keep as sunbird req and create one more req example and update the targeting criteria

priyanka-TL replied:
Do the same for programs/details, please check we can create two response example

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -2292,7 +2299,7 @@ paths:
                 $ref: '#/components/schemas/createForm400Response'
   '/scp/v1/form/update/{formId}':
-    patch:
+    post:
```
@praveenKDass update the respone of organization-extensions/createConfig and update add missing keys

priyanka-TL commented on src/api-doc/api-doc.yaml:
```diff
@@ -2564,7 +2571,9 @@ paths:
                         type: string
                       organization_code:
-                        type: number
+                        type: string
```
@praveenKDass Check the response of v1/certificates create and update api and add missing keys

priyanka-TL reviewed (commented):
reviewed on 11.50 PM

priyanka-TL committed:
Merge branch 'develop' of https://github.com/ELEVATE-Project/survey-project-creation-service into api_Doc

priyanka-TL committed:
Enhance API documentation for modules and permissions, including detailed descriptions, access control, and usage examples

priyanka-TL committed:
Enhance permission update and retrieval APIs with detailed descriptions, validation rules, and examples for improved clarity and usability

priyanka-TL committed:
Enhance role-permission mapping APIs with detailed descriptions, access control, validation rules, and examples for improved clarity and usability

priyanka-TL committed:
Enhance entity type APIs with detailed descriptions, access control, validation rules, and examples for improved clarity and usability

priyanka-TL committed:
Enhance entity creation, update, and retrieval APIs with detailed descriptions, access control, business rules, and examples for improved clarity and usability

priyanka-TL committed:
Enhance cloud services APIs with detailed descriptions, access control, and examples for generating signed and downloadable URLs, improving clarity and usability

priyanka-TL committed:
Merge branch 'develop' of https://github.com/ELEVATE-Project/survey-project-creation-service into api_Doc

priyanka-TL committed:
Update API documentation for modules and projects

- Changed header requirements from `tenant-id` and `organization-id` to `X-tenant-code` and `X-organization-code`.
- Enhanced descriptions and validation rules for module creation, updating, listing, and deletion endpoints.
- Improved clarity on project creation, update, submission, and republishing processes with detailed validation rules.
- Updated response messages and examples for better consistency and understanding.

priyanka-TL committed:
Enhance API documentation for project and reviewer endpoints with detailed descriptions, validation rules, and response examples

priyanka-TL committed:
Enhance API documentation for resource review endpoints with detailed descriptions, validation rules, and response examples

priyanka-TL committed:
Enhance API documentation for comments endpoints with detailed descriptions, validation rules, and response examples

priyanka-TL committed:
Implement feature X to enhance user experience and fix bug Y in module Z

priyanka-TL committed:
Enhance API documentation for program management endpoints with detailed descriptions, validation rules, and response examples
